paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -15,43 +15,28 @@ import androidx.fragment.app.commit
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.lifecycleScope
 import androidx.recyclerview.widget.LinearLayoutManager
-import com.stripe.android.model.StripeIntent
 import com.stripe.android.core.injection.InjectorKey
-import com.stripe.android.paymentsheet.analytics.EventReporter
+import com.stripe.android.model.StripeIntent
 import com.stripe.android.paymentsheet.databinding.FragmentPaymentsheetAddPaymentMethodBinding
 import com.stripe.android.paymentsheet.forms.FormFieldValues
-import com.stripe.android.ui.core.Amount
 import com.stripe.android.paymentsheet.model.PaymentSelection
 import com.stripe.android.paymentsheet.model.SupportedPaymentMethod
 import com.stripe.android.paymentsheet.paymentdatacollection.CardDataCollectionFragment
 import com.stripe.android.paymentsheet.paymentdatacollection.ComposeFormDataCollectionFragment
 import com.stripe.android.paymentsheet.paymentdatacollection.FormFragmentArguments
 import com.stripe.android.paymentsheet.paymentdatacollection.TransformToPaymentMethodCreateParams
-import com.stripe.android.paymentsheet.ui.AddPaymentMethodsFragmentFactory
 import com.stripe.android.paymentsheet.ui.AnimationConstants
 import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
+import com.stripe.android.ui.core.Amount
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
-internal abstract class BaseAddPaymentMethodFragment(
-    private val eventReporter: EventReporter
-) : Fragment() {
+internal abstract class BaseAddPaymentMethodFragment : Fragment() {
     abstract val viewModelFactory: ViewModelProvider.Factory
     abstract val sheetViewModel: BaseSheetViewModel<*>
 
     protected lateinit var addPaymentMethodHeader: TextView
 
-    private lateinit var selectedPaymentMethod: SupportedPaymentMethod
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        // When the fragment is destroyed and recreated, the child fragment is re-instantiated
-        // during onCreate, so the factory must be set before calling super.
-        childFragmentManager.fragmentFactory = AddPaymentMethodsFragmentFactory(
```

Comment:
If you have to re-create the fragment, then the viewModel will be created, but the SavedStateHandle can not be observed until after create is complete.